internal/backoff/backoff.go

```diff
@@ -71,3 +72,32 @@ func (bc Exponential) Backoff(retries int) time.Duration {
 	}
 	return time.Duration(backoff)
 }
+
+// RunF provides a convenient way to run a function f with a caller provided
+// backoff. It runs f repeatedly, backing off when f returns false and resetting
+// the backoff state when f returns true, until the context expires or f returns
+// a non-nil error.
+func RunF(ctx context.Context, f func() (bool, error), backoff func(int) time.Duration) {
```

Comment:
For f's signature:

Rather than return a bool and an error (that is practically ignored), how about returning only an error: if the error is `backoff.ResetBackoff`, reset the backoff; if the error is `backoff.AbortRun` (or whatever) then this function stops running `f()` and returns.

OR, maybe more generally, if it is `backoff.ResetBackoff`, reset the backoff, and if it is anything else, return from `RunF` and include that error as a return value?

Reply:
I changed `f()` to return an error value instead of `(bool, error)`. But left `RunF` as is, as I don't see the value of returning that same error from it (at least currently).